api/src/test/resources/io/strimzi/api/kafka/model/KafkaTopic-minimal.yaml

```diff
@@ -2,6 +2,3 @@ apiVersion: kafka.strimzi.io/v1beta2
 kind: KafkaTopic
 metadata:
   name: my-topic
-spec:
```

Comment:
Should the `spec` section be required?

Reply:
Hi Jakub, thanks for checking the PR. 

There is the https://github.com/strimzi/strimzi-kafka-operator/blob/main/api/src/test/java/io/strimzi/api/kafka/model/KafkaTopicCrdIT.java#L44 test that verifies resource creation with minimum parameters - now it's without replicas and partitions. I'm checking the existing tests for topic operator, possible we could add a test for topic without replicas and partitions, will come back on this one) 

As there are no other required parameters for the topic to be created we could pass an empty json as it's currently done for the Rebalance test - https://github.com/strimzi/strimzi-kafka-operator/blob/main/api/src/test/resources/io/strimzi/api/kafka/model/KafkaRebalance-minimal.yaml, but it also works without the spec field too. I guess it won't be a big difference to pass empty json or just don't pass the field at all, maybe for visibility to see it exists and can be used.

I noticed there is another issue, if I set default.replication.factor higher than number of brokers - there is no exception to provision a cluster or create a topic with default value, only if I try to use the topic it says INVALID_REPLICATION_FACTOR. Should I create a separate issue for this one or it's expected? Thanks!